### PR TITLE
Set processor affinity for snark-worker processes

### PR DESCRIPTION
**Pull request by jkrauska**
* simple script to adjust processor affinity for snark-worker processes and subthreads

### CODE REVIEW

File: scripts/utilities/tasksetter.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+cpus=$(nproc --all)
+start=0
+cores_per_process=4
+
+for pid in $(pgrep -f 'coda internal snark-worker')
+do
+    end=$((start + cores_per_process - 1))
+    taskset -a -pc $start-$end $pid
+    start=$((start + cores_per_process))
```
Comment (cmr) [inline scripts/utilities/tasksetter.sh]: maybe this is stupid but what happens if you run more snark workers than you have 4-cores available?

Comment (jkrauska) [reply scripts/utilities/tasksetter.sh]: If you have more processes than cores it just gives an error when trying to set affinity to cores that don't exist.

I decided that was ok.

eg.
`taskset: failed to set pid 33213's affinity: Invalid argument`

**Comment by enolan** [comment]
How much improvement do we get out of this? The kernel scheduler should do
a decent job. Not saying don't have a script to set it manually, just
curious.

**Comment by jkrauska** [comment]
> How much improvement do we get out of this? 

I was seeing about 10% improvement.
(10% reduction in time to perform snark work)

There's a bit of volatility in a snark worker's CPU consumption during a run. 
(1 core pegged, 4 cores pegged, cycles)

**Pull request merged**